src/rest_framework_jwt/views.py

```diff
@@ -81,6 +81,49 @@ class RefreshJSONWebTokenView(BaseJSONWebTokenAPIView):
     serializer_class = RefreshAuthTokenSerializer
 
 
+class ImpersonateJSONWebTokenView(GenericAPIView):
+    """
+    Impersonate the user by retrieving its JWT.
+
+    By default, the view permits this action only to superusers in order to
+    be backwards-compatible as much as possible. If you need to customize
+    the permission handling process, override the `permission_classes` attribute
+    using `djangorestframework`'s permission system.
+
+    Returns:
+        dict: {"token": user's JWT}
+    """
+
+    permission_classes = (IsSuperUser, )
+    serializer_class = ImpersonateAuthTokenSerializer
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+
+        serializer.is_valid(raise_exception=True)
+
+        user = serializer.validated_data.get("user")
+        payload = JSONWebTokenAuthentication.jwt_create_payload(user)
+        token = JSONWebTokenAuthentication.jwt_encode_payload(payload)
```

Comment:
What do you say we transfer this to the `ImpersonateAuthTokenSerializer`.validate`?